rs/state_manager/src/lib.rs

```diff
@@ -1496,24 +1500,36 @@ impl StateManagerImpl {
         info!(log, "Loading metadata took {:?}", starting_time.elapsed());
 
         let starting_time = Instant::now();
-        let checkpoint_heights = state_layout
-            .checkpoint_heights()
-            .unwrap_or_else(|err| fatal!(&log, "Failed to retrieve checkpoint heights: {:?}", err));
-
-        for h in checkpoint_heights {
-            let cp_layout = state_layout.checkpoint_untracked(h).unwrap_or_else(|err| {
+        // Archive unverified checkpoints.
+        let unfiltered_checkpoint_heights = state_layout
+            .unfiltered_checkpoint_heights()
+            .unwrap_or_else(|err| {
```

Comment:
If the CUP@h is not created yet, I think it is the same case because the `starting_height` will be less than h and the checkpoint@h is archived. 
In this case, it does not matter if the machine restarts due to a failure in verification or it just happens to restart during verification because we archive it both this PR and master code.

If the CUP@h is already created in the subnet and only one node restarts. If it is due to loading failure, we should archive it.

The only case different from master code is that a node happens to restart during checkpoint verification and the subnet already creates a CUP. If it happens, the replica will do some unnecessary state syncs.
The only scenario I could think of is subnet upgrade where all replicas restarts after a CUP is just created.

Reply:
I will run production tests and see if it happens frequently.

If it is indeed a frequent issue, the replica has to do verification at the restart stage and decide if the checkpoint is usable or should be archived.

Reply:
In my view unverified checkpoint is kind of tip, and we just wipe the tip at restart, so for me it's not easy to see why tip is wiped but unverified checkpoints are archived

Reply:
> why tip is wiped but unverified checkpoints are archived
I thought you were talking loading versus archiving and I was explaining why loading might not be a good option.

> In my view unverified checkpoint is kind of tip, and we just wipe the tip at restart
Speaking of archiving versus wiping, I think it is true that "unverified checkpoint is kind of tip" and we could treat them in the same way. However, since the unverified markers is a new thing, , I prefer to go with the conservative way to archive it. In the case of a verification failure, we will also have the chance to inspect it.

Reply:
that's ok, can we have a comment that it's for performance reason?

Reply:
I am not sure it is for performance reason. I would say it is for safety reason to archive it and keep it around.